### ISSUE TITLE
Some modules within crates are reported to not exist when they actually do exist.

### ISSUE THREAD

**Issue by sortofsleepy** [intellij-rust/intellij-rust#10769]
## Environment

* **IntelliJ Rust plugin version:** 0.4.199.5415-232
* **Rust toolchain version:** 1.71.0 (8ede3aae2 2023-07-12) x86_64-pc-windows-msvc
* **IDE name and version:** CLion 2023.2 (CL-232.8660.186)
* **Operating system:** Windows 10.0
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr

## Problem description
It seems for some crates, the plugin is randomly unable to parse certain modules and reports that references within those modules are are unresolved when in fact, they actually exist and should have been resolved. 

Other language parsing servers like [`rust-analyser`](https://rust-analyzer.github.io/) seem to find the missing definitions without issue.

## Steps to reproduce
I have only discovered this issue with one crate, specifically [Ash](https://crates.io/crates/ash), it's unclear whether or not this is a common issue or not but with that being said : 

* Setup a new Rust project. 
* Add the Ash crate. The latest version should exhibit this issue.
* In `main.rs` include the following 
  *`use ash::vk::CommandBuffer;`

Notes
===
`ash::vk::CommandBuffer;` should be defined within a file called `definitions.rs` within the crate. It is loaded by a file called `vk.rs`. Navigating to vk.rs shows that even `definitions.rs` appears to be missing, however, if you navigate to the actual crate files themselves, `definitions.rs` is in fact there. 

Looking through `vk.rs`, other modules within the crate appear to have been resolved, `definitions.rs` for some reason, is the only one that isn't. 

Minimal code to re-produces
===
```rust
use ash::vk::CommandBuffer;

// This constant should be resolved, just including to show how this constant is strangely found but CommandBuffer is not 
// even though they both should be included within vk.rs.
use ash::vk::MAX_DESCRIPTION_SIZE; 

fn main() {}

```

**Comment by Maksim20023** [comment]
Hi!
Thanks to the detailed description, I successfully reproduced the problem in my environment. Unfortunately, at this moment, the reason for such behavior is unclear. I forwarded this issue to our development team for investigating.
Additionally, I did a little research and found that when the ash crate is imported like this:
`ash = { git = "https://github.com/ash-rs/ash.git" }`
And it seems that all works fine.
Please check this workaround, and I'll be thankful if you leave a review about it.

**Comment by sortofsleepy** [comment]
> I forwarded this issue to our development team for investigating

👍 Sounds good, 

> Additionally, I did a little research and found that when the ash crate is imported like this:
> `ash = { git = "https://github.com/ash-rs/ash.git" }`

Ah, yes I made an edit a little later after my initial post noting that but good of you to fully explain the workaround which I should have done initially. 

Yes, using Git for now is fine for my use case, my project is just a hobby project. Hopefully this can be fixed soon!

**Comment by Voxelot** [comment]
I'm also encountering errors like this for types defined within my own workspace after updating 👎🏻
